src/VisualStudio/Core/Def/LanguageService/AbstractLanguageService`2.VsCodeWindowManager.cs

```diff
@@ -224,6 +232,94 @@ public int RemoveAdornments()
+
+                var textView = GetActiveTextView();
+                RoslynDebug.AssertNotNull(textView);
+                var textBuffer = textView.TextBuffer;
+                var isCSharpContentType = textBuffer.ContentType.IsOfType(ContentTypeNames.CSharpContentType);
+                var isVisualBasicContentType = textBuffer.ContentType.IsOfType(ContentTypeNames.VisualBasicContentType);
+
+                if (isCSharpContentType || isVisualBasicContentType)
+                {
+                    var languageServiceBroker = _languageService.Package.ComponentModel.GetService<ILanguageServiceBroker2>();
+                    var threadingContext = _languageService.Package.ComponentModel.GetService<IThreadingContext>();
+                    var asyncListenerProvider = _languageService.Package.ComponentModel.GetService<IAsynchronousOperationListenerProvider>();
+                    var asyncListener = asyncListenerProvider.GetListener("DocumentOutline");
+                    _documentOutlineView = new DocumentOutlineControl(textView, textBuffer, languageServiceBroker, threadingContext, asyncListener);
+
+                    _documentOutlineViewHost = new ElementHost
+                    {
+                        Dock = DockStyle.Fill,
+                        Child = _documentOutlineView
+                    };
+
+                    phwnd = _documentOutlineViewHost.Handle;
+                    ppCmdTarget = _documentOutlineView;
+                    Logger.Log(FunctionId.DocumentOutline_WindowOpen);
+                }
+
+                return VSConstants.S_OK;
+
+                static IWpfTextView? GetActiveTextView()
+                {
+                    var monitorSelection = (IVsMonitorSelection)Shell.Package.GetGlobalService(typeof(SVsShellMonitorSelection));
+                    if (monitorSelection == null)
+                        return null;
+
+                    if (ErrorHandler.Failed(monitorSelection.GetCurrentElementValue((uint)VSConstants.VSSELELEMID.SEID_DocumentFrame, out var curDocument)))
+                        return null;
+
+                    if (curDocument is not IVsWindowFrame frame)
+                        return null;
+
+                    if (ErrorHandler.Failed(frame.GetProperty((int)__VSFPROPID.VSFPROPID_DocView, out var docView)))
+                        return null;
+
+                    if (docView is IVsCodeWindow)
+                    {
+                        if (ErrorHandler.Failed(((IVsCodeWindow)docView).GetPrimaryView(out var textView)))
+                            return null;
+
+                        var model = (IComponentModel)Shell.Package.GetGlobalService(typeof(SComponentModel));
+                        var adapterFactory = model.GetService<IVsEditorAdaptersFactoryService>();
+                        var wpfTextView = adapterFactory.GetWpfTextView(textView);
+                        return wpfTextView;
+                    }
+
+                    return null;
+                }
```

Comment:
@jasonmalinowski for thoughts on this, and if htere's a cleaner way tknow when a roslyn view is created.

Reply:
If nothing else it's not clear to me why we're doing the dance to get the IVsCodeWindow when we already _have_ a IVsCodeWindow in this type in the first place. My worry is this means we might end up grabbing the outline for something else?

Reply:
Done - I now get the view similarly to `NavigationBarClient`